### PR TITLE
CASM 3181, CASMNET-2063, CASMNET-2039 | Aruba edge router 

### PR DESCRIPTION
**Pull request by kosonenj**
### Summary and Scope

Description:

Adds Aruba edge router to CANU as configurable option with --edge

NOTE: edge switch defaults to aruba

example of typo in --edge : 

(.venv) kosonenj@kosonenj baldar % canu generate network config --csm 1.3 -a tds --ccj baldar-ccj2.json --sls-file sls_input_file.json --edge arubas --folder generated5

Usage: canu generate network config [OPTIONS]
Try 'canu generate network config --help' for help.

Error: Invalid value for '--edge': 'arubas' is not one of 'Aruba', 'Arista'.

PR checklist (you may replace this section):

- [x] I have run `nox` locally and all tests, linting, and code coverage pass
- [ ] I have added new tests to cover the new code
- [x] My code follows the style guidelines of this project
- [x] If adding a new file, I have updated `pyinstaller.py`
- [x] I have updated the appropriate Changelog entries in `README.md`

### Issues and Related PRs

* Resolves: `CASM-3181, CASMNET-2063, CASMNET-2039`

### CODE REVIEW

File: network_modeling/models/cray-network-architecture.yaml [hunk 1]
```diff
@@ -385,9 +403,6 @@ network_v2_tds:
         - "lmem"
       shasta_name: "lmem"
       architecture_type: "river_ncn_node_4_port"
-    - lookup_name: ["viz"]
```
Comment (lukebates123) [inline network_modeling/models/cray-network-architecture.yaml:388]: Is the viz node getting deleted on accident?

File: network_modeling/models/cray-network-hardware.yaml [hunk 2]
```diff
@@ -150,13 +150,23 @@ network_hardware:
...
-  - name: "Customer Edge Router - generic stub"
-    vendor: "none"
+  - name: "arista_edge_router"
+    vendor: "arista"
     type: "switch"
     model: "customer_edge_router"
     ports:
       - count: 48
         speed: 100
+  - name: "aruba_jl636a_edge_router"
```
Comment (lukebates123) [inline network_modeling/models/cray-network-hardware.yaml:160]: Might want to use the same naming convention as the other aruba switches `- name: "Aruba 8360 JL706A"` for example.

File: network_modeling/configs/templates/1.3/aruba/sw-edge.primary.j2 [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+interface lag 1 multi-chassis
+    no shutdown
+    description sw-spine
+    no routing
+    vlan trunk native 1
+    vlan trunk allowed all
+    lacp mode active
+    lacp fallback
+    spanning-tree port-type admin-edge
+interface 1/1/1
```
Comment (lukebates123) [inline network_modeling/configs/templates/1.3/aruba/sw-edge.primary.j2:43]: assuming that these are just hardcoded for now?

**Review by lukebates123 — CHANGES_REQUESTED** [review]

File: network_modeling/models/cray-network-architecture.yaml (hunk 1 above)
Comment (kosonenj) [reply network_modeling/models/cray-network-architecture.yaml:388]: I dont think I removed it.. Maybe my file source was old and did not have it. Re-added it

File: network_modeling/configs/templates/1.3/aruba/sw-edge.primary.j2 (hunk 3 above)
Comment (kosonenj) [reply network_modeling/configs/templates/1.3/aruba/sw-edge.primary.j2]: Per feedback from luke. Deleting port configs for now. I think we should stand up a better port strategy for HSN connections (and whatever connects to edge), I.e. start reading in pt to pt tab in shcd and create jinja templates for all of these connections. Surely, we can cover these with custom configs but that approach is just prone to trial and error.

**Review by trad511 — APPROVED** [review]

**Review by lukebates123 — APPROVED** [review]

**Pull request merged**